python/ray/util/spark/cluster_init.py

```diff
@@ -529,13 +544,49 @@ def _setup_ray_cluster(
 
     cluster_unique_id = uuid.uuid4().hex[:8]
 
-    if ray_temp_root_dir is None:
-        ray_temp_root_dir = start_hook.get_default_temp_dir()
-    ray_temp_dir = os.path.join(
-        ray_temp_root_dir, f"ray-{ray_head_port}-{cluster_unique_id}"
-    )
-    os.makedirs(ray_temp_dir, exist_ok=True)
-    object_spilling_dir = os.path.join(ray_temp_dir, "spill")
+    if os.environ.get("RAY_ON_SPARK_GLOBAL_MODE", "0") == "1":
+        # global mode enabled
+        # for global mode, Ray always uses default temp dir
+        # so that local Ray client can discover it without specifying
+        # head node address.
+        if ray_temp_root_dir is not None:
+            raise ValueError(
+                "Ray on spark global mode cluster does not allow you to set "
+                "'ray_temp_root_dir' argument."
+            )
+
+        # We only allow user to launch one active Ray on spark global cluster
+        # at a time. So acquiring a global file lock before setting up a new
+        # Ray on spark global cluster.
+        global_cluster_lock_fd = os.open(
+            "/tmp/ray_on_spark_global_cluster.lock", os.O_RDWR | os.O_CREAT | os.O_TRUNC
```

Comment:
What if an user tries to start global spark cluster from a different machine, they will not see this file lock.

Reply:
This case won't happen. In Ray on spark design, we only allow calling `setup_ray_cluster` on spark driver node process (so that we can get the spark session)